src/main/java/tr/edu/ogu/ceng/service/UserService.java

```diff
@@ -55,6 +55,10 @@ public User saveUser(User user) {
 			user.setUpdateDate(dateTime);
 			user.setPassword(encodeUserPassword(user.getPassword()));
 			user.setActivity(true);
+			if (userRepository.existsById(user.getId())) {
+				log.warn("User already exists with id: {}", user.getId());
+				throw new EntityNotFoundException("User already exists!");
+			}
```

Comment:
user için update işleminde bu metot kullanılıyor olabilir mi? Bu metodun adını add olarak mı güncellemek gerekli acaba?

Reply:
@ykartal  hocam selam,
istediğiniz değişikliği sağladım yeni commit çıktım. Ama ikinci commitim birden fazla classı etkilediği için conflict oluşma olasılığı bulunmakta. Kontrol edip istediğiniz şekilde olmuş mu dönüş sağlarsanız sevinirim.

Reply:
@ykartal  hocam 3 kişiyiz bir de siz approve ederseniz güzel olur :)

Reply:
Ayrıca kastettiğim bu değildi, update işlemi için kullanan bir yer varsa oraların kontrol edilip bu metodu kullanmasını engellemelisin.